cirq/experiments/n_qubit_tomography.py

```diff
@@ -0,0 +1,194 @@
+        repetitions: number of times to sample each rotations
+        prerotations: Tuples of (phase_exponent, exponent) parameters for gates
+            to apply to the qubits before measurement. The actual rotation
+            applied will be
+                cirq.PhasedXPowGate(
+                    phase_exponent=phase_exponent, exponent=exponent)
+            If none, we use [(0, 0), (0, 0.5), (0.5, 0.5)], which corresponds to
+            rotation gates [I, X**0.5, Y**0.5]
+    Returns:
+        Tomography result which contains the density matrix of the qubits
+        determined by tomography.
+    """
+    if prerotations == None:
+        prerotations = [(0, 0), (0, 0.5), (0.5, 0.5)]
+
+    exp = StateTomographyExperiment(qubits, prerotations)
+    rot_moment, rot_sweep = (exp.rot_moment, exp.rot_sweep)
+    probs = get_state_tomography_data(sampler,
+                                      qubits,
+                                      circuit,
+                                      rot_moment,
+                                      rot_sweep,
+                                      repetitions=repetitions)
+    return exp.fit_density_matrix(probs)
+
+
+def get_state_tomography_data(sampler: cirq.Sampler,
+                              qubits: List[cirq.Qid],
+                              circuit: cirq.Circuit,
+                              rot_moment: cirq.Circuit,
+                              rot_sweep: cirq.Sweep,
+                              repetitions: int = 1000) -> np.ndarray:
+    """Gets the data for each rotation string added to the circuit.
+
+    For each sequence in prerotation_sequences gets the probability of all
+    2**n bit strings.  Resulting matrix will have dimensions
+    (len(rot_sweep)**n, 2**n).
+    This is a default way to get data that can be replaced by the user if they
+    have a more advanced protocol in mind.
+
+    Args:
+        sampler: sampler to collect the data from.
+        qubits: qubits to do the tomography on.
+        circuit: circuit to do the tomography on.
+        rot_moment: moment with parametrized rotation gates to do before the
+            final measurements.
+        rot_sweep: The list of rotations on the qubits to perform before
+            measurement.
+        repetitions: number of times to sample each rotation sequence.
+
+    Returns:
+        2D array of probabilities, where first index is which pre-rotation was
+        applied and second index is the qubit state.
+    """
+    results = sampler.run_sweep(circuit + rot_moment +
+                                [cirq.measure(*qubits, key='z')],
+                                params=rot_sweep,
+                                repetitions=repetitions)
+
+    all_probs = []
+    for result in results:
+        hist = result.histogram(key='z')
+        probs = [hist[i] for i in range(2**len(qubits))]
+        all_probs.append(np.array(probs) / repetitions)
+    return np.array(all_probs)
```

Comment:
Nit: This could be refactored to not copy so many arrays but I don't think this is a bottleneck.